src/Nri/Ui/Carousel/V2.elm

```diff
@@ -0,0 +1,175 @@
+module Nri.Ui.Carousel.V2 exposing
+    ( viewWithCombinedControls, viewWithPreviousAndNextControls
+    , viewWithTabControls, AriaLabel(..)
+    )
+
+{-| Patch changes:
+
+  - added new carousel APIs (with tabbed controls/previous and next controls, and combined controls)
+
+@docs viewWithCombinedControls, viewWithPreviousAndNextControls
+@docs viewWithTabControls, AriaLabel
+
+-}
+
+import Accessibility.Styled.Aria as Aria
+import Css exposing (..)
+import Html.Styled as Html exposing (..)
+import Html.Styled.Attributes as Attrs exposing (css)
+import TabsInternal.V2 as TabsInternal
+
+
+{-| Builds a carousel with previous and next controls
+Returns:
+`controls`: previous and next buttons element
+`slides` the container with the carousel contents
+`containerAttributes` attributes that should be used on the parent div of both the button and slides elements
+-}
+viewWithPreviousAndNextControls :
+    { selected : id
+    , panels :
+        List
+            { id : id
+            , slideHtml : Html msg
+            , ariaLabel : AriaLabel
+            }
+    , viewPreviousButton : Html msg
+    , viewNextButton : Html msg
+    , ariaLabel : AriaLabel
+    , controlListStyles : List Style
+    }
+    ->
+        { controls : Html msg
+        , slides : Html msg
+        , containerAttributes : List (Attribute msg)
+        }
+viewWithPreviousAndNextControls config =
+    { controls =
+        div [ css config.controlListStyles ]
```

Comment:
Growth's use case requires rendering one button on each side of the panel, and for our case we'll want to add the "__ of __" label somewhere in between them. Should we return both buttons separately rather than in a container rendered by ourselves?

<img width="93" alt="Screenshot 2023-08-25 at 15 17 48" src="https://github.com/NoRedInk/noredink-ui/assets/753421/40d0c851-2c0a-4991-9ecc-f3fc0ed6ec7e">

<img width="1037" alt="Screenshot 2023-08-25 at 15 15 51" src="https://github.com/NoRedInk/noredink-ui/assets/753421/f93bb6b9-7019-48ee-b350-2e6fcb9551ca">

Reply:
Would definitely be weird to receive the buttons' html just to return it untouched, though? 😆 

💭  Kind of cheating, but this made me wonder if we want to turn `viewPreviousButton` and `viewNextButton` into something like:

```elm
{  ...
  , previousButton : { name : String, attributes: List Button.Attribute }
  , nextButton : { name : String, attributes: List Button.Attribute } 
  , ...
}
```

We could then call `Button.button` ourselves, ensuring that actual buttons are being used (which is also suggesded by the spec). We could even add some default-but-overridable "Previous/Next slide" labels too, while we are at it.

Reply:
I don't know if we should use Nri.UI.Button, they have some styles by default, and the whole component is unystlyed by default.

Reply:
Interesting!

Curious to know what the bats think, but I think the value we are shooting for is **not imposing** anything wrt styles. I personally don't see a practical issue with the button having _some_ style by default (as long as it can be customize to accommodate all use cases we care about).

On the other hand, I just looked at how we the Carousel in the landing page is implemented currently, and we are [using clickable divs with svgs inside](https://github.com/NoRedInk/NoRedInk/blob/49b58a5d4534925c60872f666f98cb9c8f259485/contento/ui/src/Nri/LandingPage.elm#L2931-L2948) (when I'm pretty sure it would be better to [use customized buttons as we do in GDs](https://github.com/NoRedInk/NoRedInk/blob/49b58a5d4534925c60872f666f98cb9c8f259485/monolith/ui/src/Page/Learn/GuidedDrafts/ShortResponseWriting/WritingSample.elm#L763-L776)). Makes me think that there actually is value in enforcing buttons in the API.

WDYT? Happy to let a bat weigh in before we make any decisions!

Reply:
I don't have strong opinions on this one, but I think it's worth to have a opinion from someone on the accessibility bats team first. Maybe @tesk9 could help us on this point?

Reply:
Ended up enforcing ClickableSvg! Marking this thread as resolved.